### PR TITLE
Updated doc for decimal support

### PR DESCRIPTION
**Pull request by viadea**
Updated doc for decimal support

### CODE REVIEW

File: docs/supported_ops.md [hunk 1]
```diff
@@ -14,11 +14,11 @@ apply to other versions of Spark, but there may be slight changes.
...
 # General limitations
 ## `Decimal`
-The `Decimal` type in Spark supports a precision
-up to 38 digits (128-bits). The RAPIDS Accelerator in most cases stores values up to
-64-bits and will support 128-bit in the future. As such the accelerator currently only
-supports a precision up to 18 digits. Note that
-decimals are disabled by default in the plugin, because it is supported by a relatively
+The `Decimal` type in Spark supports a precision up to 38 digits (128-bits). 
+The RAPIDS Accelerator supports 128-bit starting from version 21.12 and decimals are enabled by default.
+Please check this [Decicmal Support](compatibility.md#decimal-support) for more details.
```
Comment (sameerz) [inline docs/supported_ops.md:19]:
```suggestion
Please check  [Decimal Support](compatibility.md#decimal-support) for more details.
```

File: docs/supported_ops.md [hunk 2]
```diff
@@ -14,11 +14,11 @@ apply to other versions of Spark, but there may be slight changes.
...
+
+Prior to 21.12 decimals are disabled by default in the plugin, because it is supported by a relatively
 small number of operations presently. This can result in a lot of data movement to and
 from the GPU, slowing down processing in some cases.
 Result `Decimal` precision and scale follow the same rule as CPU mode in Apache Spark:
```
Comment (sameerz) [inline docs/supported_ops.md:24]:
```suggestion
`Decimal` precision and scale follow the same rule as CPU mode in Apache Spark:
```

File: docs/supported_ops.md [hunk 3]
```diff
@@ -14,11 +14,11 @@ apply to other versions of Spark, but there may be slight changes.
...
-64-bits and will support 128-bit in the future. As such the accelerator currently only
-supports a precision up to 18 digits. Note that
-decimals are disabled by default in the plugin, because it is supported by a relatively
+The `Decimal` type in Spark supports a precision up to 38 digits (128-bits). 
+The RAPIDS Accelerator supports 128-bit starting from version 21.12 and decimals are enabled by default.
+Please check this [Decicmal Support](compatibility.md#decimal-support) for more details.
+
+Prior to 21.12 decimals are disabled by default in the plugin, because it is supported by a relatively
 small number of operations presently. This can result in a lot of data movement to and
 from the GPU, slowing down processing in some cases.
```
Comment (sameerz) [inline docs/supported_ops.md:23]: I suggest removing this section entirely.  
```suggestion
```

**Review by sameerz — CHANGES_REQUESTED** [review]
The `supported_ops.md` file is generated from https://github.com/NVIDIA/spark-rapids/blob/branch-22.02/sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala .  I made some recommendations on the changes, please consider them for changes in that file.

**Comment by sameerz** [comment]
Removing `[skip-ci]` since this change should touch more than documentation.

**Comment by viadea** [comment]
> Removing `[skip-ci]` since this change should touch more than documentation.

Sure changed the `TypeChecks.scala` for the same thing as well

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala [hunk 4]
```diff
@@ -1706,14 +1706,11 @@ object SupportedOpsDocs {
...
-    println("The `Decimal` type in Spark supports a precision")
-    println("up to 38 digits (128-bits). The RAPIDS Accelerator in most cases stores values up to")
-    println("64-bits and will support 128-bit in the future. As such the accelerator currently only")
-    println(s"supports a precision up to ${DType.DECIMAL64_MAX_PRECISION} digits. Note that")
-    println("decimals are disabled by default in the plugin, because it is supported by a relatively")
-    println("small number of operations presently. This can result in a lot of data movement to and")
-    println("from the GPU, slowing down processing in some cases.")
-    println("Result `Decimal` precision and scale follow the same rule as CPU mode in Apache Spark:")
+    println("The `Decimal` type in Spark supports a precision up to 38 digits (128-bits). ")
+    println("The RAPIDS Accelerator supports 128-bit starting from version 21.12 and decimals are enabled by default.")
```
Comment (sameerz) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala:1710]: Do we have a limit of 100 characters in a line?

Comment (viadea) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala]: Fixed

**Review by revans2 — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Pull request merged**